### PR TITLE
Java architect L5 DZ

### PR DESCRIPTION
**Pull request by D1mkaGit**
DZ5:
Command pattern - "абстрактная" генерация репортов, "абстрактным" пользователем
State pattern - реализация "абстракного" (без наваротов) логирования времени работником (работа/отдых)

### CODE REVIEW

File: JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/AccountingReport.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package command;
+
+public class AccountingReport extends Report {
+
+    public AccountingReport(Reports reports) {
+        super(reports);
+    }
+
+    @Override
+    public void Generate() {
```
Comment (aimiebell) [inline JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/AccountingReport.java:10]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/ClientsReport.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package command;
+
+public class ClientsReport extends Report {
+
+    public ClientsReport(Reports reports) {
+        super(reports);
+    }
+
+    @Override
+    public void Generate() {
```
Comment (aimiebell) [inline JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/ClientsReport.java:10]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/GoodsReport.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package command;
+
+public class GoodsReport extends Report {
+
+    public GoodsReport(Reports reports) {
+        super(reports);
+    }
+
+    @Override
+    public void Generate() {
```
Comment (aimiebell) [inline JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/GoodsReport.java:10]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/ReportI.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package command;
+
+public interface ReportI {
+    void Generate();
```
Comment (aimiebell) [inline JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/ReportI.java:4]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/User.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    final ReportI genGoodsRep;
+
+    public User(ReportI genClientsRep, ReportI genAccountingRep, ReportI genGoogsRep) {
+        this.genClientsRep = genClientsRep;
+        this.genAccountingRep = genAccountingRep;
+        this.genGoodsRep = genGoogsRep;
+    }
+
+    void genClientsRep() {
+        genClientsRep.Generate();
```
Comment (aimiebell) [inline JavaArchitecturesAndDesignPatterns/patterns/behavioral/src/main/java/command/User.java:15]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Review by dikkini — APPROVED** [review]

**Pull request merged**